gremlin-go/driver/driverRemoteConnection.go

```diff
@@ -80,13 +85,36 @@ func NewDriverRemoteConnection(
 		transporterType: settings.TransporterType,
 		logHandler:      logHandler,
 		connection:      connection,
+		Session:         settings.Session,
 	}
 
 	return &DriverRemoteConnection{client: client}, nil
 }
 
 // Close closes the DriverRemoteConnection.
 func (driver *DriverRemoteConnection) Close() error {
+	// If DriverRemoteConnection has spawnedSessions then they must be closed as well.
+	if len(driver.spawnedSessions) > 0 {
+		driver.client.logHandler.logf(Info, closingSpawnedSessions, driver.client.url)
+		flag := false
+		for _, session := range driver.spawnedSessions {
+			err := session.Close()
+			if err != nil {
+				driver.client.logHandler.logf(Error, closeSessionError, err)
+				flag = true
+			}
+		}
+		if flag {
+			return errors.New("there was at least one error while closing spawned sessions. Check error log for more info")
+		}
+		return nil
+	}
+
+	if driver.isSessionBound() {
+		driver.client.logHandler.logf(Info, closeDRCSession, driver.client.url, driver.client.Session)
+	} else {
+		driver.client.logHandler.logf(Info, closeDRC, driver.client.url)
+	}
 	return driver.client.Close()
```

Comment:
1. If the driver has spawned sessions, then I'm assuming that `isSessionBound` check at the bottom here is supposed to trigger, which it will not because the function is returning in the loop above.
2. Since apparently your DRC can be live as well as have spawned sessions, then perhaps missing `driver.client.Close()` in the case where spawned sessions exist is a bug.

Reply:
`isSessionBound` returns `True` if the DRC is a session, not if it has spawnedSessions. Perhaps I should rename it to just `isSession`

Reply:
Regardless, the client is not closing if there are spawned sessions.